tests/system/feature_store/test_spark_engine.py

```diff
@@ -2208,8 +2199,79 @@ def test_as_of_join_result(self):
         )
         spark_engine_res = resp.to_dataframe().sort_index(axis=1)
 
-        self._logger.info(f"result of LOCAL engine merger:\n  {local_engine_res}")
-        self._logger.info(f"result of SPARK engine merger:\n  {spark_engine_res}")
-
         assert spark_engine_res.shape == (2, 2)
         assert local_engine_res.equals(spark_engine_res)
+
+    @pytest.mark.parametrize(
+        "timestamp_for_filtering",
+        [None, "other_ts", "bad_ts", {"fs1": "other_ts"}, {"fs1": "bad_ts"}],
+    )
+    @pytest.mark.parametrize("passthrough", [True, False])
+    def test_time_filter(self, timestamp_for_filtering, passthrough):
+        test_base_time = datetime.fromisoformat("2020-07-21T12:00:00")
+
+        df = pd.DataFrame(
+            {
+                "ent": ["a", "b", "c", "d"],
+                "ts_key": [
+                    test_base_time - pd.Timedelta(minutes=1),
+                    test_base_time - pd.Timedelta(minutes=2),
+                    test_base_time - pd.Timedelta(minutes=3),
+                    test_base_time - pd.Timedelta(minutes=4),
+                ],
+                "other_ts": [
+                    test_base_time - pd.Timedelta(minutes=4),
+                    test_base_time - pd.Timedelta(minutes=3),
+                    test_base_time - pd.Timedelta(minutes=2),
+                    test_base_time - pd.Timedelta(minutes=1),
+                ],
+                "val": [1, 2, 3, 4],
+            }
+        )
+
+        base_path = self.test_output_subdir_path(url=False)
+        path = f"{base_path}/df_for_filter.parquet"
+
+        fsys = fsspec.filesystem(
+            "file" if self.run_local else v3iofs.fs.V3ioFS.protocol
+        )
+        fsys.makedirs(base_path, exist_ok=True)
+        df.to_parquet(path=path, filesystem=fsys)
+        source = ParquetSource("pq1", path=path)
+
+        fset1 = fstore.FeatureSet(
+            "fs1", entities=["ent"], timestamp_key="ts_key", passthrough=passthrough
+        )
+        self.set_targets(fset1, also_in_remote=True)
+
+        fstore.ingest(fset1, source)
+
+        vec = fstore.FeatureVector("vec1", ["fs1.val"])
+
+        target = ParquetTarget(
+            "mytarget", path=f"{self.output_dir()}-get_offline_features"
+        )
+
+        resp = fstore.get_offline_features(
+            feature_vector=vec,
+            start_time=test_base_time - pd.Timedelta(minutes=3),
+            end_time=test_base_time,
+            timestamp_for_filtering=timestamp_for_filtering,
+            engine="spark",
+            run_config=fstore.RunConfig(local=self.run_local),
```

Comment:
Need to fix as per #3646.
```suggestion
            run_config=fstore.RunConfig(local=self.run_local, kind="remote-spark"),
```